src/main/java/com/couchbase/lite/replicator/PullerInternal.java

```diff
@@ -845,63 +845,45 @@ public void run() {
 
     @Override
     public void changeTrackerCaughtUp() {
-        workExecutor.submit(new Runnable() {
-            @Override
-            public void run() {
-                try {
-                    Log.d(Log.TAG_SYNC, "changeTrackerCaughtUp");
-                } catch (Exception e) {
-                    e.printStackTrace();
-                    throw new RuntimeException(e);
-                }
-            }
-        });
-
+        Log.e(Log.TAG_SYNC, "changeTrackerCaughtUp");
         // for continuous replications, once the change tracker is caught up, we
         // should try to go into the idle state.
         if (isContinuous()) {
-
             // this has to be on a different thread than the replicator thread, or else it's a deadlock
             // because it might be waiting for jobs that have been scheduled, and not
             // yet executed (and which will never execute because this will block processing).
             new Thread(new Runnable() {
                 @Override
                 public void run() {
-
                     try {
-
                         if (batcher != null) {
-                            Log.d(Log.TAG_SYNC, "batcher.waitForPendingFutures()");
+                            // if batcher delays task execution, need to wait same amount of time. (0.5 sec or 0 sec)
+                            try { Thread.sleep(batcher.delayToUse()); }catch(Exception e){}
+                            Log.e(Log.TAG_SYNC, "batcher.waitForPendingFutures()");
                             batcher.waitForPendingFutures();
                         }
```

Comment:
Problem is that inbox in Batcher is not empty, but pendingFeatures in Batcher is empty. Calling `batcher.waitForPendingFutures();` before `Feature` is put in pendingFeatures does not make sense. 

We might be able to change `Batcher.waitForPendingFutures()` method to wait `inbox.size()` becomes 0 before calling `ScheduledFuture.get() `.

What do you think?